tools/platyPS/ValidateHelp.ps1

```diff
@@ -0,0 +1,179 @@
+﻿[CmdletBinding()]
+Param(
+    [Parameter(Mandatory=$True, Position=0)]
+    [String]$Service,
+    [Parameter(Mandatory=$True, Position=1)]
+    [String]$PathToRepo
+)
+
+$HelpPath = "$PathToRepo\src\ResourceManager\$Service\Commands.$Service\help"
+
+# Check to see if the given service has any markdown help files
+if (!(Test-Path $HelpPath))
+{
+    Write-Host "The service $Service does not contain any markdown help files"
+}
+else
+{
+    # Keep track of errors in the markdown
+    [String[]]$errors = @()
+
+    # Get all of the markdown help files
+    $files = Get-ChildItem -Path $HelpPath
+
+    foreach ($file in $files)
+    {
+        # Ignore the module page
+        if ($file.Name -eq "AzureRM.$Service.md")
+        {
+            continue
+        }
+
+        $fileErrors = @()
+
+        # For each help file, check to see if they are missing help in any section
+        $content = Get-Content $HelpPath\$file
+        
+        # Iterate over each line in the file
+        for ($idx = 0; $idx -lt $content.Length; $idx++)
+        {
+            switch ($content[$idx])
+            {
+                "## SYNOPSIS"
+                {
+                    $foundSynopsis = $False
+                    $idx++
+
+                    # Check each line between SYNOPSIS and SYNTAX for any text
+                    for (;;)
+                    {
+                        $foundSynopsis = $foundSynopsis -or (!([string]::IsNullOrWhiteSpace("$($content[$idx])")))
+
```

Comment:
You might consider using a regular expression and the C# Regex class for each of these.  You can see examples of usage in the PublishModules script.